src/devices/Rtc/RtcBase.cs

```diff
@@ -10,15 +10,57 @@ namespace Iot.Device.Rtc
     /// </summary>
     public abstract class RtcBase : IDisposable
     {
+        private TimeZoneInfo _timeZone;
+
         /// <summary>
-        /// The Device's <see cref="System.DateTime"/>
+        /// The Device's raw <see cref="System.DateTime"/>.
+        /// The caller must be aware of the device's time zone.
+        /// The behavior of the <see cref="System.DateTime.Kind"/> property is implementation-dependent (typically it is ignored)
         /// </summary>
-        public virtual DateTime DateTime
+        public virtual DateTime RtcDateTime
         {
             get => ReadTime();
             set => SetTime(value);
         }
 
+        /// <summary>
+        /// Set or retrieves the current date/time. This property returns a <see cref="DateTimeOffset"/> and
+        /// is therefore correct regardless of the current time zone (when <see cref="TimeZone"/> is set correctly).
+        /// </summary>
+        public DateTimeOffset DateTime
+        {
+            get
+            {
+                var now = RtcDateTime;
+                return new DateTimeOffset(now.Ticks, TimeZone.GetUtcOffset(now));
+            }
+            set
+            {
+                var clockNow = new DateTime((value.UtcDateTime + TimeZone.GetUtcOffset(value)).Ticks, DateTimeKind.Local);
```

Comment:
I kinda feel like there should be two time zones:
- RTC time zone (hidden from user, perhaps accepted in the constructor, if not provided it should be the same as local time zone)
- local time zone

when setting this property RTC time zone and local time zone should become the same and therefore RTC time should be directly set to whatever the local time is. If we leave as is I'm sure there will be many edge cases

Reply:
i.e. say I'm in UTC time zone and I move the clock to to UTC+1, as long as I provide correct time zone everything should work fine (RTC time zone is UTC, local is UTC+1), the moment I try to explicitly set time I'd expect RTC time zone to be now UTC+1 and internally work in the UTC+1 rather than UTC

Reply:
I was trying to keep it simple. Having two time zones is complex enough, adding a third (TZ the class returns, TZ the clock is in and UTC) will be mind-blowing for some people. I also don't see a reason why you should set the RTC to something else than your local TZ or utc. 

Additionally, if we want to do that, it would need to be in subclasses, because we can't set the timezone of the system clock, as we expect that after setting it, `DateTime.Now` will return the correct time.